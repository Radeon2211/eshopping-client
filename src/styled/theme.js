const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
