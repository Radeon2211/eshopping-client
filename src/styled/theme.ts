const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#777',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#06AD96',
    greenDark: '#009680',
    red: '#F22613',
    redLight: '#F94D2A',
    redDark: '#985A5A',
  },
  shadows: {
    level1: '0 0 3px rgba(0, 0, 0, 0.03)',
    level2: '0 0 3px rgba(0, 0, 0, 0.3)',
    level3: '0 0 5px rgba(0, 0, 0, 0.3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Noto Sans, sans-serif',
    text: 'Nunito, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
    level5: 100000,
  },
  fontSizes: {
    level1: '1.3rem',
    level2: '1.4rem',
    level3: '1.5rem',
    level4: '1.7rem',
    level5: '2.2rem',
    level6: '2.6rem',
  },
  durations: {
    level1: 0.1,
    level2: 0.2,
    level3: 0.3,
  },
  lineHeights: {
    level1: '1',
    level2: '1.1',
    level3: '1.2',
    level4: '1.3',
    level5: '1.4',
    level6: '1.5',
  },
};

export type ThemeSpacing = keyof typeof theme.spacings;
export type ThemeZIndex = keyof typeof theme.zIndexes;
export type ThemeFontSize = keyof typeof theme.fontSizes;
export type ThemeDuration = keyof typeof theme.durations;
export type ThemeLineHeight = keyof typeof theme.lineHeights;

export default theme;
