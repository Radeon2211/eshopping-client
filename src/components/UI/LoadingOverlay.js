import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Loader from './Loader';

const SC = {};
SC.Wrapper = styled.div`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.light1Transparent};
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  justify-items: ${({ alignLoader }) => alignLoader};
  left: 0;
  padding-top: ${({ theme }) => theme.spacings.level5};
  position: absolute;
  top: 0;
  width: 100%;
  z-index: ${({ theme }) => theme.zIndexes.level1};

  & .info {
    font-size: ${({ theme }) => theme.fontSizes.level2};
    margin-top: ${({ theme }) => theme.spacings.level2};
  }

  ${({ zeroPadding }) => {
    if (zeroPadding) {
      return `
        padding: 0;
      `;
    }
    return ``;
  }}
`;

const LoadingOverlay = (props) => {
  const { alignLoader, loaderSize, zeroPadding, disableText } = props;

  const [isInfoVisible, setIsInfoVisible] = useState(false);

  useEffect(() => {
    const timeout = setTimeout(() => {
      setIsInfoVisible(true);
    }, 8000);
    return () => clearTimeout(timeout);
  }, [setIsInfoVisible]);

  let info = false;
  if (isInfoVisible && !disableText) {
    info = <span className="info">Just a second</span>;
  }

  return (
    <SC.Wrapper alignLoader={alignLoader} zeroPadding={zeroPadding}>
      <Loader size={loaderSize} />
      {info}
    </SC.Wrapper>
  );
};

LoadingOverlay.defaultProps = {
  zeroPadding: false,
  alignLoader: 'center',
  disableText: false,
  loaderSize: '',
};

LoadingOverlay.propTypes = {
  zeroPadding: PropTypes.bool,
  alignLoader: PropTypes.string,
  loaderSize: PropTypes.string,
  disableText: PropTypes.bool,
};

export default LoadingOverlay;
