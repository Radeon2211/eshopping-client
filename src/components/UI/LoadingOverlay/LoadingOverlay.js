import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Loader from '../Loader/Loader';

const SC = {};
SC.Wrapper = styled.div`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.light1Transparent};
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  left: 0;
  padding-top: ${({ theme }) => theme.spacings.level5};
  position: absolute;
  top: 0;
  width: 100%;
  z-index: ${({ theme }) => theme.zIndexes.level1};

  & .info {
    font-size: ${({ theme }) => theme.fontSizes.level2};
    margin-top: ${({ theme }) => theme.spacings.level2};
  }

  ${({ alignLoader }) => {
    if (alignLoader === 'top') {
      return `
        justify-items: start;
      `;
    }
    if (alignLoader === 'center') {
      return `
        justify-items: center;
      `;
    }
    return ``;
  }}

  ${({ zeroPadding }) => {
    if (zeroPadding) {
      return `
        padding: 0;
      `;
    }
    return ``;
  }}
`;

const LoadingOverlay = (props) => {
  const { alignLoader, loaderSize, zeroPadding } = props;

  const [isInfoVisible, setIsInfoVisible] = useState(false);

  useEffect(() => {
    const timeout = setTimeout(() => {
      setIsInfoVisible(true);
    }, 8000);
    return () => clearTimeout(timeout);
  }, [setIsInfoVisible]);

  let info = false;
  if (isInfoVisible) {
    info = <span className="info">Just a second</span>;
  }

  return (
    <SC.Wrapper alignLoader={alignLoader} zeroPadding={zeroPadding}>
      <Loader size={loaderSize} />
      {info}
    </SC.Wrapper>
  );
};

LoadingOverlay.defaultProps = {
  zeroPadding: false,
};

LoadingOverlay.propTypes = {
  zeroPadding: PropTypes.bool,
  alignLoader: PropTypes.string.isRequired,
  loaderSize: PropTypes.string.isRequired,
};

export default LoadingOverlay;
