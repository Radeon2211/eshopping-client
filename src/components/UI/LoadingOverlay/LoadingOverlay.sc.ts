import styled from 'styled-components';
import { LoadingOverlayProps } from './LoadingOverlay';

export const Wrapper = styled.div<{ $alignLoader: LoadingOverlayProps['alignLoader'] }>`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.light1Transparent};
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  justify-items: ${({ $alignLoader }) => $alignLoader};
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: ${({ theme }) => theme.zIndexes.level1};
`;
