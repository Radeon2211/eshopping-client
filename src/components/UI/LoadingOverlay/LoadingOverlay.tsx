import { useState, useEffect } from 'react';
import Loader from '../Loader/Loader';
import PlainText from '../PlainText';
import * as SC from './LoadingOverlay.sc';

export interface LoadingOverlayProps {
  alignLoader?: string;
  loaderSize?: 'big' | 'medium' | 'small';
  disableText?: boolean;
}

export default function LoadingOverlay({
  alignLoader = 'center',
  loaderSize = 'medium',
  disableText = false,
}: LoadingOverlayProps) {
  const [isInfoVisible, setIsInfoVisible] = useState<boolean>(false);

  useEffect(() => {
    const timeout = setTimeout(() => {
      setIsInfoVisible(true);
    }, 8000);
    return () => clearTimeout(timeout);
  }, [setIsInfoVisible]);

  return (
    <SC.Wrapper $alignLoader={alignLoader} data-testid="LoadingOverlay">
      <Loader size={loaderSize} />
      {isInfoVisible && !disableText && (
        <PlainText $size="level2" $mgTop="level2" data-testid="LoadingOverlay-info">
          Just a second
        </PlainText>
      )}
    </SC.Wrapper>
  );
}
