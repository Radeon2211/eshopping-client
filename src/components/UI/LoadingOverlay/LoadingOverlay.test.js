import React from 'react';
import { render, cleanup, screen, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';
import theme from '../../../styled/theme';
import LoadingOverlay from './LoadingOverlay';

const setUp = (disableText = false) => {
  return render(
    <ThemeProvider theme={theme}>
      <LoadingOverlay disableText={disableText} />
    </ThemeProvider>,
  );
};

afterEach(cleanup);

describe('<LoadingOverlay />', () => {
  beforeEach(() => {
    jest.useFakeTimers('modern');
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  describe('check how renders', () => {
    it('should render without info', () => {
      setUp();
      expect(screen.getByTestId('LoadingOverlay')).toBeInTheDocument();
    });

    it('should render info after 8 seconds', () => {
      setUp();
      expect(screen.queryByTestId('LoadingOverlay-info')).not.toBeInTheDocument();
      act(() => {
        jest.advanceTimersByTime(8000);
      });
      expect(screen.getByTestId('LoadingOverlay-info')).toBeInTheDocument();
    });

    it('should NOT render info after 7999ms', () => {
      setUp();
      act(() => {
        jest.advanceTimersByTime(7999);
      });
      expect(screen.queryByTestId('LoadingOverlay-info')).not.toBeInTheDocument();
    });

    it('should NOT render info after 8 seconds if disableText is true', () => {
      setUp(true);
      act(() => {
        jest.advanceTimersByTime(8000);
      });
      expect(screen.queryByTestId('LoadingOverlay-info')).not.toBeInTheDocument();
    });
  });
});
