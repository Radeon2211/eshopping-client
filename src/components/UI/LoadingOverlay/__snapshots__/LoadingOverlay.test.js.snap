// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingOverlay /> Check how renders should render correctly without info 1`] = `
<DocumentFragment>
  <div
    class="sc-hKgILt kaHYlz"
    data-testid="LoadingOverlay"
  >
    <div
      class="sc-gsTCUz Ckrks"
      data-testid="Loader"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
